kong/plugins/acl/schema.lua

```diff
@@ -16,6 +16,7 @@ return {
               type = "array",
               elements = { type = "string" }, }, },
           { hide_groups_header = { type = "boolean", required = true, default = false, description = "If enabled (`true`), prevents the `X-Consumer-Groups` header from being sent in the request to the upstream service." }, },
+          { always_use_authenticated_groups = { type = "boolean", required = true, default = false, description = "If enabled (`true`), the authenticated groups will always be used even when an authenticated consumer already exists. If the authenticated groups don't exist, it will fallback to use the groups associated with the consumer. By default the authenticated groups will only be used when there is no consumer or the consumer is anonymous." } },
```

Comment:
Which part of the code implements the logic of:

> it will fallback to use the groups associated with the consumer

and is it covered in the test?

Reply:
Yes. The part of the else block. https://github.com/Kong/kong/pull/13184/files#diff-6a2485484a2746e4da00c5a27533d8d45662e48bdf453dd7e596ae5e53e4854dR94